HeterogeneousCore/CUDACore/interface/ESProduct.h

```diff
@@ -19,23 +20,24 @@ namespace cms {
     class ESProduct {
     public:
       ESProduct() : gpuDataPerDevice_(numberOfDevices()) {
+        cms::cuda::ScopedSetDevice scopedDevice;
```

Comment:
This needs to be protected for 0 devices, I'll prepare a fix.

On the other hand, I need to re-educate myself why exactly this happens.

Reply:
> On the other hand, I need to re-educate myself why exactly this happens.

Seems that the `PixelCPEFast` ESProduct is used for both GPU modules and CPU modules, and it holds `cms::cuda::ESProduct<...>` as a member. This should really be refactored, but it might be most efficient to leave that to the time we have a pattern for issuing the CUDA operations from the ESProducer instead of the ESProduct (hopefully later this fall).